src/shapes/Cone.cpp

```
@@ -1,46 +1,136 @@
-  double t_plane_A = -oc_dot_normal / dir_dot_normal;
-  double t_plane_B = (_height - oc_dot_normal) / dir_dot_normal;
-
-  double t3 = std::min(t_plane_A, t_plane_B);
-  double t4 = std::max(t_plane_A, t_plane_B);
-
-  double final_t_start = std::max(t1, t3);
-  double final_t_end = std::min(t2, t4);
-
-  if (final_t_start < final_t_end && final_t_end >= 0) {
-    return {final_t_start, _color, this};
+Math::Vector3D Raytracer::Cone::getNormal(const Math::Point3D &hit_point) const {
```

Cppcheck (2.10-2)

[style] The function 'getNormal' is never used.

(unusedFunction)